src/main/java/iudx/resource/server/metering/MeteringServiceImpl.java

```diff
@@ -1,330 +1,242 @@
 
-    if (validationCheck!=null && validationCheck.containsKey(ERROR)){
-      responseBuilder =
-          new ResponseBuilder(FAILED).setTypeAndTitle(400).setMessage(validationCheck.getString(ERROR));
-      handler.handle(Future.failedFuture(responseBuilder.getResponse().toString()));
-      return this;
+        return this;
     }
-    request.put(TABLE_NAME, databaseTableName);
-    query = queryBuilder.buildCountQuery(request);
-
-    LOGGER.trace(query);
-    LOGGER.debug("Info: Query constructed: " + query.getString(QUERY_KEY));
 
-    Future<JsonObject> countResult = executeCountQuery(query);
-    countResult.onComplete(
-        countResultHandler -> {
-          if (countResultHandler.succeeded()) {
-            if (request.getString(HEADER_OPTIONS) != null)
-              handler.handle(Future.succeededFuture(countResultHandler.result()));
-            else {
-              int totalCount= Integer.parseInt(
-                  countResultHandler.result().getJsonArray(RESULTS).getJsonObject(0).getString(TOTAL));
-              if(totalCount>=10000){
-                responseBuilder =
-                    new ResponseBuilder(FAILED).setTypeAndTitle(400).setMessage(RESPONSE_LIMIT_EXCEED);
-                handler.handle(Future.failedFuture(responseBuilder.getResponse().toString()));
-              }
-              else {
-                query = queryBuilder.buildReadingQuery(query);
-                query.put(COUNT,totalCount);
-                Future<JsonObject> initialReadResult = executeReadQuery(query);
-                initialReadResult.onComplete(initialReadHandler->{
-                  if(initialReadHandler.succeeded()){
-                    Future<JsonObject> remainingReadResult = executeRemainingReadQuery(query);
-                      remainingReadResult.onComplete(remainingReadHandler->{
-                        if(remainingReadHandler.succeeded()){
-                          JsonArray jsonArray = remainingReadHandler.result().getJsonArray(RESPONSE_ARRAY);
-                          responseBuilder =
-                              new ResponseBuilder(SUCCESS).setTypeAndTitle(200).setTotalHits(totalCount).setData(jsonArray);
-                          handler.handle(Future.succeededFuture(responseBuilder.getResponse()));
-                        }
-                        else{
-                          LOGGER.info("FAILED "+remainingReadHandler.cause());
-                        }
-                      });
-                  }
-                });
-              }
+    private void countQuery(JsonObject request, Handler<AsyncResult<JsonObject>> handler) {
+        queryCount = queryBuilder.buildCountReadQueryByPG(request);
+        Future<JsonObject> resultCountPg = databaseOperation(queryCount);
+        resultCountPg.onComplete(countHandler -> {
+            if (countHandler.succeeded()) {
+                total = Integer.parseInt(countHandler.result().getJsonArray("result").getJsonObject(0).getString("count"));
```

Comment:
This may fail if there is no field by the name `results` or `count` in the response array

Reply:
SOLVED